src/main/java/com/pubnub/api/endpoints/presence/SetState.java

```diff
@@ -131,4 +155,18 @@ protected boolean isAuthRequired() {
         return true;
     }
 
+    @Nullable
+    private String getStateParamValue() throws PubNubException {
+        Map<String, Object> stateParamValue = new HashMap<>();
+        for (String channel : channels) {
+            stateParamValue.put(channel, state);
+        }
+        for (String channelGroup : channelGroups) {
+            stateParamValue.put(channelGroup, state);
+        }
+
+        String stringifiedStatePerChannel = this.getPubnub().getMapper().toJson(stateParamValue);
+        String encodedStatePerChannel = PubNubUtil.urlEncode(stringifiedStatePerChannel);
+        return encodedStatePerChannel;
```

Comment:
What should happen if there's no channels or channelGroups yet? I don't think sending empty state is what the user would expect